### DISCUSSION TITLE
Older program getting excellent new results. TilingZoeDepth

### DISCUSSION THREAD

**Discussion by gituser123456789000** [nagadomi/nunif#101] [Ideas]
https://github.com/n8ventures/TilingZoeDepth_GUI

Here's a GUI fork of TilingZoeDepth that was easy for me to get running locally. 3.0.2 can be downloaded directly at around 1.5GB or so and there is an application that you go into and when you run a model, it will download that model the first time. Very easy.

To get 3.1.0 running, I wasn't able to figure out how to install whatever was needed, so I simply combined the 3.1.0 files into the complete 3.0.2 folder.

It now supports 32bit mode, which gives amazing detail.. no BS. I knew about it previously, way before this GUI, and I was less knowledgeable back then. Now I was able to get it running and test better with it.

I'd like to see it do tiling for Depth-Anything, Marigold, Metric3D.. Tiling is a game changer.

Here's one result from a test image, but I'm getting great results all around. Some or many 32bit maps may look like they have waves/banding defects, but this isn't visible when used to convert in 3DCombine. I assume 3DCombine clears up the issues.

Tiled ZoeD_N model worked best for this image (other models supported are just ZoeD_K and ZoeD_NK).

Anaglyph, crosseye, and SBS attached

![anaglyph 8](https://github.com/nagadomi/nunif/assets/156575428/4670b3d4-f0c4-4a13-9246-019605cd269d)
![cross 8](https://github.com/nagadomi/nunif/assets/156575428/dd66a913-eff1-45a1-85a8-a6878eff7584)
![SBS 8](https://github.com/nagadomi/nunif/assets/156575428/6e572baa-5593-45b6-87b5-7f95c39591a3)

**Comment by kaelsonofkrypto** [comment]
Wow this does look great, especially on the SBS, oh yes I second this please add this if possible

**Comment by nagadomi** [comment]
This can be supported.
My concerns are,
1. At least 10x slower  (1x + 3x3 tiled). If more hierarchical resolution is used, it is 20x or 40x slower. It may be hard to use for video. (very time consuming)
2. There are better methods such as BoostingMonocularDepth. However, they cannot be used due to license issue.

I can develop a similar method myself, but if it is hard to use for video, I am not very motivated.

**Reply by donrikk** [reply]
if ur willing to try to make ur method we would be happy to test it if it isnt viable then no problem, i would be interested in seeing if ur method could present a faster variant than its native counterpart

**Reply by gituser123456789000** [reply]
I agree, it does take long and wouldn't be feasible for full length movies. But I'm getting consistently excellent results from 32bit Tiling. It would be an excellent addition for images. Or very short clips.

And this is only for ZoeDepth. I imagine results could be significantly even better if there were also options for Tiling with Depth-Anything, Marigold, Metric3D..

And I also like that you mentioned even higher levels of Tiling ("If more hierarchical resolution is used").. if I understood that correctly. That would also be a great option to have. Again, not for full movies with the technology of today, but an insanely detailed 3D still image in a minute or few minutes is not a big wait time, especially for important, sentimental photos, family, friends. This wait time is nothing.

**Reply by gituser123456789000** [reply]
I've done Tiling with other models and the results also showed great promise. I did it manually by dividing images into 8 for example, running each of the segments through Depth-Anything, then recombining them manually.

This didn't fix all of the obvious edges not lining up that the program does fix, but the proof of concept showed that other models will pull out more details. Upscaling also pulled out additional details. I was doing tests with 8x upscale and 8x manual tiling.

**Reply by gituser123456789000** [reply]
The time-frames don't seem completely crazy. I would spend 10x more time for potentially much higher quality. Even a full movie, some settings I use would process a nearly 2 hour movie in only around 2 ½ hours... So even 10x longer is still only 1 day to process with potentially much higher quality depth detail. Definitely interested to see what Tiling can do. It would also be cool if we can choose custom levels of tiling, as you mentioned. I'm going to start looking at TilingZoeDepth GUI fork to see if I can edit it to do further tiling or get other models running with it, but I'm not a coder, so I don't think I'll get too far.

**Reply by gituser123456789000** [reply]
We have a newer program that's able to tile Depth-Anything v2.. and in my first direct comparison, 4x4 Tiled Giant looked better (at least in some ways) than regular 518.

I'm bumping this to request Tiling again.

My plan was to see how GenPercept maps looked when actually converted, so I cherry-picked one of the best looking GenPercept results I had and compared it to iw3 (unofficial) Giant 518, iw3 (unofficial) Giant 1080p res (the original image res), 4x4 Tiled Giant, and 4x4 Tiled Large

The depth maps were compared using 3rd party 3DCombine to give a fair result. 4x4 Tiled Giant may very well look the best. It's a give and take on which details look best between 4x4 Tiled Giant, GenPercept, and regular 518 res. Tiled Large did not look as good as those 3.. and 1080p res Giant looked the worst because of degradation of accuracy.

This was just 1 test image, but showed Tiling has its place and the result may turn out to be the best.

The Tiling program I used was limited to 4x4, 8x8, etc tiling at the time.. but options for 2x2 and 6x6 are being added soon. 2x2 Tiling I'd expect to perform well also, or potentially tunr out to be consistently the best.. as it would be more detailed that default 518, and have less cloudiness caused by masking and recombining the tiles.

I'll share examples when the tiling program gets updated with the 2x2 tiling option and I'm able to use it again.

**Comment by gituser123456789000** [comment]
Also... especially after we've already had some speed increases (performance fixes), and we also learned we only have to use 518 resolution,.. and it was mentioned that Windows could be getting a significant performance boost again when a certain dependency in the program receives an expected update... Tiling suddenly now would be similar to the speeds we were getting without tiling just a short time ago.
